webapp/channels/src/components/admin_console/access_control/editors/cel_editor/editor.tsx

```diff
@@ -0,0 +1,363 @@
+            run: validateSyntax,
+        });
+
+        monaco.editor.addKeybindingRule({
+            keybinding: monaco.KeyMod.Alt | monaco.KeyCode.Enter,
+            command: VALIDATE_POLICY_SYNTAX_COMMAND_ID,
+        });
+
+        return () => {
+            if (monacoRef.current) {
+                monacoRef.current.dispose();
+                monacoRef.current = null;
+            }
+        };
+    }, []);
+
+    return (
+        <div className={`cel-editor ${className}`}>
+            <MonacoLanguageProvider schemas={schemas}/>
+
+            <div
+                className='cel-editor__container'
+                data-status-color={editorState.statusBarColor}
+            >
+                {!editorState.expression && (
+                    <div
+                        className='policy-editor-placeholder'
+                        aria-label='CEL Expression Editor'
+                    >
+                        {placeholder}
+                    </div>
+                )}
+
+                <div
+                    ref={editorRef}
+                    className='cel-editor__input'
+                />
+                <div
+                    className='cel-editor__status-bar'
+                    style={{backgroundColor: editorState.statusBarColor}}
+                    onClick={() => {
+                        if (!editorState.isValidating && editorState.validationErrors.length === 0 &&
+                            !(editorState.isValid && editorState.statusBarColor === 'var(--online-indicator)')) {
+                            validateSyntax();
+                        }
+                    }}
+                    role='button'
+                    tabIndex={0}
+                    onKeyDown={(e) => {
+                        if (e.key === 'Enter' || e.key === ' ') {
+                            if (!editorState.isValidating && editorState.validationErrors.length === 0 &&
+                                !(editorState.isValid && editorState.statusBarColor === 'var(--online-indicator)')) {
+                                validateSyntax();
+                            }
+                        }
+                    }}
+                    data-validation-state={
+                        (() => {
+                            if (editorState.isValidating) {
+                                return 'validating';
+                            }
+
+                            if (editorState.validationErrors.length > 0) {
+                                return 'error';
+                            }
+
+                            if (editorState.isValid && editorState.statusBarColor === 'var(--online-indicator)') {
+                                return 'validated';
+                            }
+
+                            return 'unvalidated';
+                        })()
+                    }
+                >
+                    <div className='cel-editor__status-message'>
+                        {(() => {
```

Comment:
(Post-merge) Could you pull the status bar out into its own component and remove the function call nested in it? Currently, we have JSX which contains a function call which contains multiple returns which each contain more JSX (one which contains a ternary statement which contains even more JSX), so it's a bit too complex and deeply nested 😅